tests/ulyanov_et_al_2016/test_loss.py

```diff
@@ -1,32 +1,155 @@
+    encoder = multi_layer_encoder.extract_encoder(layer)
+    target_repr = gram_matrix(encoder(target_image), normalize=True)
+    input_repr = gram_matrix(encoder(input_image), normalize=True)
+
+    configs = ((True, True, 1.0), (False, False, input_repr.size()[0]))
+    for impl_params, normalize_by_num_channels, extra_batch_normalization in configs:
+        with subtests.test(impl_params=impl_params):
+            intern_target_repr = (
+                target_repr / target_repr.size()[-1]
+                if normalize_by_num_channels
+                else target_repr
+            )
+            intern_inüut_repr = (
+                input_repr / input_repr.size()[-1]
+                if normalize_by_num_channels
+                else input_repr
+            )
+            op = loss.UlyanovEtAl2016GramOperator(encoder, impl_params=impl_params,)
+            op.set_target_image(target_image)
+            actual = op(input_image)
+
+            score = mse_loss(intern_inüut_repr, intern_target_repr,)
+            desired = score / extra_batch_normalization
+
+            assert actual == ptu.approx(desired, rel=1e-3)
+
+
+def test_ulyanov_et_al_2016_style_loss(subtests):
+    for impl_params, instance_norm, stylization in itertools.product(
+        (True, False), (True, False), (True, False)
+    ):
+        style_loss = loss.ulyanov_et_al_2016_style_loss(
             impl_params=impl_params,
-            multi_layer_encoder=multi_layer_encoder,
-            layer=layer,
+            instance_norm=instance_norm,
+            stylization=stylization,
+        )
+        assert isinstance(style_loss, ops.MultiLayerEncodingOperator)
+
+        with subtests.test("encoding_ops"):
+            assert all(
+                isinstance(op, loss.UlyanovEtAl2016GramOperator)
+                for op in style_loss.operators()
+            )
+
+        layers, layer_weights = zip(
+            *[(op.encoder.layer, op.score_weight) for op in style_loss.operators()]
+        )
+        with subtests.test("layers"):
+            desired_layers = (
+                {"relu1_1", "relu2_1", "relu3_1", "relu4_1"}
+                if impl_params and instance_norm
+                else {"relu1_1", "relu2_1", "relu3_1", "relu4_1", "relu5_1"}
+            )
+            assert set(layers) == desired_layers
+
+        with subtests.test("layer_weights"):
+            desired_layer_weights = (1e0,) * len(desired_layers)
+            assert layer_weights == pytest.approx(desired_layer_weights)
+
+        with subtests.test("score_weight"):
+            if impl_params:
+                if instance_norm:
+                    score_weight = 1e0
+                else:
+                    score_weight = 1e3 if stylization else 1e0
+            else:
+                score_weight = 1e0
+            assert style_loss.score_weight == pytest.approx(score_weight)
+
+
+def test_ulyanov_et_al_2016_perceptual_loss(subtests):
+    for stylization in (True, False):
```

Comment:
For now you can ignore all cases with `stylization is False`. We probably want to change the API since it leads to errors such as #20. The discussion of this is out of the scope of this PR though.